python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -0,0 +1,293 @@
+            **settings.prepare_settings_dict(),
+        )
+        response_metadata = self._get_metadata_from_response(response)
+
+        return [self._create_chat_message_content(response, choice, response_metadata) for choice in response.choices]
+
+    async def get_streaming_chat_message_contents(
+        self,
+        chat_history: ChatHistory,
+        settings: AzureAIInferenceChatPromptExecutionSettings,
+        **kwargs: Any,
+    ) -> AsyncGenerator[list[StreamingChatMessageContent], Any]:
+        """Get streaming chat message contents from the Azure AI Inference service.
+
+        Args:
+            chat_history: A list of chats in a chat_history object.
+            settings: Settings for the request.
+            kwargs: Optional arguments.
+
+        Returns:
+            A list of chat message contents.
+        """
+        response: AsyncStreamingChatCompletions = await self.client.complete(
+            stream=True,
+            messages=self._format_chat_history(chat_history),
+            model_extras=settings.extra_parameters,
+            **settings.prepare_settings_dict(),
+        )
+
+        async for chunk in response:
+            if len(chunk.choices) == 0:
+                continue
+            chunk_metadata = self._get_metadata_from_response(chunk)
+            yield [
+                self._create_streaming_chat_message_content(chunk, choice, chunk_metadata) for choice in chunk.choices
+            ]
+
+    def _get_metadata_from_response(self, response: ChatCompletions | AsyncStreamingChatCompletions) -> dict[str, Any]:
+        """Get metadata from the response.
+
+        Args:
+            response: The response from the service.
+
+        Returns:
+            A dictionary containing metadata.
+        """
+        return {
+            "id": response.id,
+            "model": response.model,
+            "created": response.created,
+            "usage": response.usage,
+        }
+
+    def _create_chat_message_content(
+        self, response: ChatCompletions, choice: ChatChoice, metadata: dict[str, Any]
+    ) -> ChatMessageContent:
+        """Create a chat message content object.
+
+        Args:
+            response: The response from the service.
+            choice: The choice from the response.
+            metadata: The metadata from the response.
+
+        Returns:
+            A chat message content object.
+        """
+        items = []
+        if choice.message.tool_calls:
+            for tool_call in choice.message.tool_calls:
+                items.append(
+                    FunctionCallContent(
+                        id=tool_call.id,
+                        name=tool_call.function.name,
+                        arguments=tool_call.function.arguments,
+                    )
+                )
+
+        return ChatMessageContent(
+            role=AuthorRole(choice.message.role),
+            items=items,
+            content=choice.message.content,
```

Comment:
it's more efficient if the content is already put into a TextContent here, and then all items (one or more) are passed in.